pytentiostat/main.py

```diff
@@ -3,19 +3,68 @@
 from pytentiostat.reporter import save_data_to_file
 from pytentiostat.tester import experiment
 from pytentiostat.routines import startup_routine, closing_routine
+from pytentiostat.plotter import plot_updater, plot_initializer
+import matplotlib.pyplot as plt
+import sys
+import time
+
+
+class BoardCom:
+    def __init__(self):
+        self.com, self.board, self.pin_a0, self.pin_a2, self.pin_d9 = startup_routine()
+
+    def configure_board(self):
+        self.com, self.board, self.pin_a0, self.pin_a2, self.pin_d9 = startup_routine()
+
 
 config_data = parse_config_file()
 
-# Initialize Experiment
-com, board, a0, a2, d9 = startup_routine()
+try:
+    # Initialize Experiment
+    board_instance = BoardCom()
+    # com, board, a0, a2, d9 = startup_routine()
+except KeyboardInterrupt:
+    sys.exit("Stopped. Make any changes and then restart the software.\nExiting...")
+
+board_objects = (board_instance.pin_a0, board_instance.pin_a2, board_instance.pin_d9)
+
+reconfig = "sting"
+
+while True:
+    try:
+        # Run the experiment and get the config_data
+        times, voltages, currents = experiment(config_data, *board_objects)
+        break
+    except KeyboardInterrupt:
+        reconfig = input("Experiment stopped. To restart the same experiment, press Enter. \n"
+                         "If you want to do a different experiment, type \"New\"\n"
+                         "If you need to reconnect the poteniostat, type \"Reconnect\" \n")
+        if reconfig == "":
+            reconfig = "string"
+            plt.close()
+            continue
+        elif reconfig.lower() == "new":
+            reconfig = "string"
+            plt.close()
+            config_data = parse_config_file()
+            continue
+        elif reconfig.lower() == "reconnect":
+            reconfig = "string"
+            plt.close()
+            closing_routine(board_instance.board, board_instance.pin_d9)
+            board_instance.configure_board()
+            board_objects = (board_instance.pin_a0, board_instance.pin_a2, board_instance.pin_d9)
+            continue
+        elif reconfig == "string":
+            break
 
-# Run the experiment and get the config_data
-board_objects = (board, a0, a2, d9)
-times, voltages, currents = experiment(config_data, *board_objects)
+try:
+    # Generate a config_data report
+    collected_data = zip(times, voltages, currents)
+    save_data_to_file(config_data, collected_data)
+except KeyboardInterrupt:
+    sys.exit("Interrupted while saving!")
```

Comment:
do you want some cleanup here?  close the file and delete it maybe?  Not sure what state the filesystem is left things are interrupted mid-save.

Reply:
This one also feels awkward to handle. I don't imagine anyone trying to interrupt during a save but I put it there to handle just in case.

Reply:
....so don't bother this this.   Less code = less to maintain!  But if you wrap the write in a ``with open`` statement I think it will handle it....that is why we use that structure in part I think.

Reply:
The save_data_to_file function uses `with open` when saving. Perhaps I'll just let that handle any interrupt rather than write our own exception.